util/flags/diagnostics.go

```diff
@@ -55,7 +56,7 @@ func AddDiagnosticsOptions(fs *pflag.FlagSet, options *DiagnosticsOptions) {
 }
 
 // GetDiagnosticsOptions returns metrics options which can be used to configure a Manager.
-func GetDiagnosticsOptions(options DiagnosticsOptions) metricsserver.Options {
+func GetDiagnosticsOptions(options DiagnosticsOptions, tlsOptionOverrides []func(*tls.Config)) metricsserver.Options {
```

Comment:
Note:

Like this it is a breaking change, which is OK I think. This way we enforce users to remember this PR and pass in the `tlsOptionoverrides`.

However, I think we can't cherry-pick it like this.

We could though use:

```suggestion
func GetDiagnosticsOptions(options DiagnosticsOptions, tlsOptionOverrides ...func(*tls.Config)) metricsserver.Options {
```

instead. So if folks bump to a new patch version, it won't break their current setup. But people may forget to pass through the tls options then 🤔

Reply:
Happy to do this - thank you for the suggestion!